lib/auth0/api/v2/log_streams.rb

```diff
@@ -0,0 +1,82 @@
+module Auth0
+  module Api
+    module V2
+      # Methods to use the log streams endpoints
+      module LogStreams
+        attr_reader :log_streams_path
+
+        # Retrieves a list of all log streams.
+        # @see https://auth0.com/docs/api/management/v2#!/Log_Streams/get_log_streams
+        # @return [json] Returns the log streams.
+        def log_streams()
+          get(log_streams_path)
+        end
+        alias get_log_streams log_streams
+
+        # Retrieves a log stream by its ID.
+        # @see https://auth0.com/docs/api/management/v2#!/Log_Streams/get_log_streams_by_id
+        # @param id [string] The id of the log stream to retrieve.
+        #
+        # @return [json] Returns the log stream.
+        def log_stream(id)
+          raise Auth0::InvalidParameter, 'Must supply a valid log stream id' if id.to_s.empty?
+          path = "#{log_streams_path}/#{id}"
+          get(path)
+        end
+        alias get_log_stream log_stream
+
+        # Creates a new log stream according to the JSON object received in body.
+        # @see https://auth0.com/docs/api/management/v2#!/Log_Streams/post_log_streams
+        # @param name [string] The name of the log stream.
+        # @param type [string] The type of log stream
+        # @param options [hash] The Hash options used to define the log streams's properties.
+        #   * :httpEndpoint [string] The name of the log stream. Must contain at least one character.
+        #   * :httpContentType [string] The algorithm used to sign tokens.
+        #   * :httpContentFormat [string] The secret used to sign tokens when using symmetric algorithms.
+        #   * :httpAuthorization [integer] The amount of time (in seconds) that the token will be valid after being issued.
```

Comment:
These options are specific to the HTTP WebHook log stream type, other types of log streams will have different properties. Maybe best to refer to the API2 docs for the available options instead of trying to capture them all?